core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -1235,6 +1248,47 @@ public void testUpdateSortOrder() {
         sortedByX.sortOrder().fields().get(0).nullOrder());
   }
 
+  @Test
+  public void testStatistics() {
+    Schema schema = new Schema(Types.NestedField.required(10, "x", Types.StringType.get()));
+
+    TableMetadata meta =
+        TableMetadata.newTableMetadata(
+            schema, PartitionSpec.unpartitioned(), null, ImmutableMap.of());
+    Assert.assertEquals(
+        "Should default to no statistics files", ImmutableList.of(), meta.statisticsFiles());
+  }
+
+  @Test
+  public void testSetStatistics() {
+    Schema schema = new Schema(Types.NestedField.required(10, "x", Types.StringType.get()));
+
+    TableMetadata meta =
+        TableMetadata.newTableMetadata(
+            schema, PartitionSpec.unpartitioned(), null, ImmutableMap.of());
+
+    Assert.assertSame(
+        "Should detect no statistics to remove",
+        meta,
+        TableMetadata.buildFrom(meta).removeStatistics(42L).build());
```

Comment:
I don't think this really fits in a `testSetStatistics` method since you're testing the default behavior of `removeStatistics`. I'd probably move this to a different test method. I also don't see a test to check that `removeStatistics` works without removing other stats files.

It would also be good to have a test to validate replacement behavior, since `setStatistics` is idempotent.

Reply:
> I don't think this really fits in a `testSetStatistics` method since you're testing the default behavior of `removeStatistics`. I'd probably move this to a different test method

moved

> I also don't see a test to check that removeStatistics works without removing other stats files.

added

> It would also be good to have a test to validate replacement behavior, since setStatistics is idempotent.

added